pymc3/distributions/distribution.py

```diff
@@ -211,16 +211,101 @@ class DensityDist(Distribution):
 
     """
 
-    def __init__(self, logp, shape=(), dtype=None, testval=0, random=None, *args, **kwargs):
+    def __init__(
+        self,
+        logp,
+        shape=(),
+        dtype=None,
+        testval=0,
+        random=None,
+        wrap_random_with_dist_shape=True,
+        check_shape_in_random=True,
+        *args,
+        **kwargs
+    ):
+        """
+        Parameters
+        ----------
```

Comment:
NICE!

Reply:
The one question I have -- and this is not a question about your change -- is whether we should have a default `shape` that is not `None` (i.e.,  that records the fact that the programmer did not specify shape).
I'm sure that the shape *is* most likely `()`, but worried about the case where it isn't, and the programmer/modeler has not realized that they need to specify the shape.
But this is a different issue, and should not delay merging this PR.

Reply:
I would say no, we shouldn't add a special empty shape value. Ideally, the default random variable's shape should be deduced from its parameters. We are far from this goal. At the moment the shape information must be explicitly supplied to every distribution in pymc3, not just `DensityDist`. The default behavior package wise is to assume a scalar distribution.
When we start to move towards a more symbolic handling of shapes, we should address whether to reserve a special empty value or not.